tests/app/test_create_app.py

```diff
@@ -0,0 +1,76 @@
+"""Verifies that instance_links are being retrieved properly from LINKS. Verifies that app_data.json.j2
+contains the instance link information"""
+from unittest import mock
+import requests_mock
+import json
+import foremast.app.create_app
+from foremast.app import SpinnakerApp
+
+
+@mock.patch('foremast.app.create_app.LINKS', new={"test1": "https://test1.com", "test2": "https://test2.com"})
+def test_instance_links():
+    """Checks if instance_links is being populated properly"""
+    pipeline_config = {
+        "instance_links": {
+            "test3": "https://test2.com",
+            "test4": "https://test4.com",
+        }
+    }
+    spinnaker_app = SpinnakerApp(pipeline_config=pipeline_config)
+    instance_links = spinnaker_app.retrieve_instance_links()
+    assert instance_links == {"test1": "https://test1.com"}, "Instance Links are not being retrieved properly"
+
+    pipeline_config = { "instance_links": {} }
+    spinnaker_app = SpinnakerApp(pipeline_config=pipeline_config)
+    instance_links = spinnaker_app.retrieve_instance_links()
+    assert instance_links == {"test1": "https://test1.com", "test2": "https://test2.com"}, "Instance Links are not being retrieved properly"
+
+
+@mock.patch('foremast.app.create_app.API_URL', new='https://spinnaker.build.gogoair.com:7777')
```

Comment:
This should not default to a real URL. Unittests should work without internet access and we dont want to show our internal URLs on the open source code.